llvm/lib/CodeGen/MIRPrinter.cpp

```diff
@@ -576,6 +576,9 @@ void MIRPrinter::convertCallSiteObjects(yaml::MachineFunction &YMF,
       printRegMIR(ArgReg.Reg, YmlArgReg.Reg, TRI);
       YmlCS.ArgForwardingRegs.emplace_back(YmlArgReg);
     }
+    // Get type id.
+    if (CSInfo.second.TypeId)
+      YmlCS.TypeId = CSInfo.second.TypeId->getZExtValue();
```

Comment:
This would be nicer w/ C++17 destructuring. That's probably a bit invasive for this patch since you'd have to change a bunch of unrelated code, but is something that could land as a separate NFC PR relatively quickly.